src/oba_from_uniswap/main.py

```diff
@@ -0,0 +1,227 @@
+# Postprocesses csv files obtained from dune analytics with routed trades
+
+import csv
+import argparse
+import ast
+import json
+from tqdm import tqdm
+from functools import lru_cache
+from frozendict import frozendict
+from fractions import Fraction as F
+from diskcache import Cache
+
+from ..subgraph import GraphQLClient, UniswapClient
+
+uniswap = UniswapClient()
+disk_cache = Cache(directory='.cache')
+
+def load_swaps(filename):
+    """Parse a csv file to a list of dicts."""
+    r = []
+    with open(filename, "r") as f:
+        reader = csv.reader(f)
+        first = True
+        for row in reader:
+            # Skip header.
+            if first:
+                first = False
+                continue
+            (block_number, index, sell_amount,
+            buy_amount, path, output_amounts) = row
+
+            path = ast.literal_eval(path)
+            path = ['0x' + address for address in path]
+            output_amounts = ast.literal_eval(output_amounts.replace('L',''))
+            output_amounts = list(map(int, output_amounts))
+            r.append({
+                'block_number': int(block_number),
+                'index': int(index),
+                'sell_amount': int(sell_amount),
+                'buy_amount': int(buy_amount),
+                'path': path,
+                'output_amounts': output_amounts
+            })
+    return r
+
+
+def load_tokens(filename):
+    with open(filename, "r") as f:
+        return json.load(f)
+
+
+def filter_swaps(swaps, tokens):
+    return [
+        s for s in swaps
+        if {s['path'][0], s['path'][-1]} <= tokens
+    ]
+
+def split_by_pair(orders):
+    pairs = {(o['sellToken'], o['buyToken']) for o in orders}
+    return  {
+        pair: [o for o in orders if {o['sellToken'], o['buyToken']} == set(pair)]
+        for pair in pairs
+    }
+
+
+# This decorator caches this call parameters and result on disk, so
+# that we don't have to wait each time we re-run the code for the same
+# list of swaps (happens a lot while developing).
+@disk_cache.memoize()
+def get_token_info(swaps):
+    token_info = {}
+    for swap in tqdm(swaps, desc='Querying token info from thegraph'):
```

Comment:
Just curious what `tqdm()` does!?

Reply:
It shows a nice progress bar on the terminal!